tools/pylint-extensions/pylint-guidelines-checker/pylint_guidelines_checker.py

```diff
@@ -371,15 +374,23 @@ def visit_functiondef(self, node):
         :type node: ast.FunctionDef
         :return: None
         """
+ 
         try:
-            if node.parent.name.endswith("Client") and node.is_method() and not node.name.startswith("_") and \
-                    node.parent.name not in self.ignore_clients:
-                if node.args.kwarg and "azure.core.tracing.decorator.distributed_trace" not in node.decoratornames() \
-                        and "builtins.classmethod" not in node.decoratornames():
-                    self.add_message(
-                        msgid="client-method-missing-tracing-decorator", node=node, confidence=None
-                    )
-        except AttributeError:
+            path = node.root().name
+            split_path = path.split(".")
+            new_path = ".".join(split_path[:len(split_path)-1])
+            if new_path.count("_") == 0:
+                if node.parent.name.endswith("Client") and node.is_method() and not node.name.startswith("_") and \
+                        node.parent.name not in self.ignore_clients:
+                    if node.args.kwarg and node.name not in self.ignore_functions and not node.name.endswith("client"):
+                        if self.ignore_decorators[0] not in node.decoratornames() \
+                            and self.ignore_decorators[1] not in node.decoratornames() \
+                                and self.ignore_decorators[2] not in node.decoratornames():
+                        
```

Comment:
This syntax isn't maintainable. You probably want to do something like take the union of `decoratornames` and `ignore_decorators` and verify that union is empty. That way if you add to the list you don't need to modify this.